llvm-project/stage1.cmake

```diff
@@ -0,0 +1,45 @@
+# TODO: comment every one of these.
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
+set(CMAKE_CXX_COMPILER "/usr/bin/clang++" CACHE FILEPATH "")
+set(CMAKE_C_COMPILER "/usr/bin/clang" CACHE FILEPATH "")
+set(CMAKE_EXE_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
+set(CMAKE_SHARED_LINKER_FLAGS "-fuse-ld=lld" CACHE STRING "")
+set(LLVM_DEFAULT_TARGET_TRIPLE "x86_64-alpine-linux-musl" CACHE STRING "")
+set(LLVM_ENABLE_LLD ON CACHE BOOL "")
+set(LLVM_ENABLE_PROJECTS "clang;lld;" CACHE STRING "")
+set(LLVM_ENABLE_RUNTIMES "compiler-rt;libcxx;libcxxabi;libunwind;" CACHE STRING "")
+set(LLVM_ENABLE_ZLIB "FORCE_ON" CACHE STRING "")
+set(LLVM_TARGETS_TO_BUILD "X86;" CACHE STRING "")
```

Comment:
I didn't know about `host`! Neat!.  I think we should strive to keep as much of this host agnostic as possible, since we likely will want to re-bootstrap different host architectures; i.e. I think a realistic goal in the near term is to produce aarch64 host binaries.  I plan to get an M2 or M1 mac soon.

For the target triple, I don't we can (or want to) run shell scrips from .cmake files.  I *think* what we could do is have the docker image run `clang -print-multiarch` then use the result as the value of a `-D` flag passed in to the cmake invocation.  IDK if that would work; I haven't tried it yet, but I think it could.

I will make this change you suggested, and try my `-D $(clang -print-multiarch)` idea.

Reply:
yeah, this seems to work.

Reply:
> yeah, this seems to work.

Spoke too soon.  stage 1 `install-compiler-rt` will fail with a cryptic:
>>   Host compiler does not support '-fuse-ld=lld'

I'll drop my `-D` change; that can be sorted out later, maybe closer to when we have an idea for making builds for other host architectures.

Reply:
Sure. Realistically, if we are only going to push builds through CI, there is not much point in worrying about other architectures until GitHub Actions has native arm64 support. We can build arm64 images through GitHub Actions but they use `qemu-user`, which will be horrendously slow for our use case.

I am hoping my arm64 workstation will ship here soon, then I can start playing with it locally at least.

Reply:
looks like I just need `llvm-ar` in stage2. That's...weird.

Reply:
done. PTAL